### ISSUE TITLE
investigate beta diversity parallelization using sklearn.metrics.pairwise.pairwise_distances

### ISSUE THREAD

**Issue by gregcaporaso** [biocore/scikit-bio#1185]
We might be able to [call this](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.pairwise_distances.html) instead of ``pdist`` internally and use the scikit-learn parallel framework. This should be separate from #1040 though, as that one is already very big and this wouldn't involve an API change.

**Comment by wasade** [comment]
It doesn't look like this is doing the block parallelization, but is instead parallelizing over rows similar to how the partitioning was done in QIIME. The code is being passed through to joblib but is partitioning only over a single axis (see [here](https://github.com/scikit-learn/scikit-learn/blob/c957249/sklearn/metrics/pairwise.py#L1060) and the preceding line).  

The downsides of this strategy for fast unifrac is that all samples need to be resident. The impact is that the counts array requires a substantial memory footprint (estimated at 3.2TB for the EMP open reference). Additionally, it is not possible to prune the input tree down leading to performance overhead.

**Comment by gregcaporaso** [comment]
related to #1181

**Comment by gregcaporaso** [comment]
@wasade, true that there are likely better approaches. with this hooked up (see my [PR](https://github.com/biocore/scikit-bio/pull/1194)) we'll have working parallel beta diversity now which we can use as our baseline during development of other approaches.

**Comment by jairideout** [comment]
Fixed in #1194.
